### DISCUSSION TITLE
clock tree synthesis

### DISCUSSION THREAD

**Discussion by Fahim36201** [The-OpenROAD-Project/OpenROAD-flow-scripts#868] [Q&A]
How do i set max transition limit for cts also is there a way to set max insertion delay?

**Comment by vijayank88** [comment]
Example sdc [file](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/designs/sky130hd/aes/constraint.sdc)
Check examples [here](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/designs/sky130hd/microwatt/constraint.sdc)

**Reply by Fahim36201** [reply]
Thanks @vijayank88 . are there any other ways without modifying the sdc file like any command or attribute? for example in i want to insertion delay, max transition in cts in that case?

**Reply by vijayank88** [reply]
It is depending on how you're executing tool. If interactive execution, you can update constraints whenever need. For automated flow, everything should defined .sdc file before execution of the flow.

**Reply by vijayank88** [reply]
> Thanks @vijayank88 . are there any other ways without modifying the sdc file like any command or attribute? for example in i want to insertion delay, max transition in cts in that case?

Run `gcd` design and explore `results` for each PnR stage .sdc file generates. Just update .sdc and re-run again with updated constraint file

**Reply by Fahim36201** [reply]
> It is depending on how you're executing tool. If interactive execution, you can update constraints whenever need. For automated flow, everything should defined .sdc file before execution of the flow.

Yes i am trying to automate it.

**Reply by Fahim36201** [reply]
> > Thanks @vijayank88 . are there any other ways without modifying the sdc file like any command or attribute? for example in i want to insertion delay, max transition in cts in that case?
> 
> Run `gcd` design and explore `results` for each PnR stage .sdc file generates. Just update .sdc and re-run again with updated constraint file

thanks @vijayank88. will try it out.
